### DISCUSSION TITLE
Motion commander: Controlling two Crazyflies with one python script

### DISCUSSION THREAD

**Discussion by Vrinda73** [bitcraze/discussions#458] [Q&A] [answered]
Hello! 
I am a graduate student and I needed some help with controlling two CFs simultaneously. I have two radios. I went through the swarm.py code in cflib. As per my understanding it requires mocap or position estimates. However I just plan to use the flow sensor and give velocity setpoints through start_linear_motion(vx,vy,vz,0) using the motion commander class for each CF. I will like to receive some guidance regarding this.
Thank you for your time!

**Comment by krichardsson** [comment, marked as answer]
Hi!

Actually mocap is not needed, but good position estimation will make it easier for the controller in the CF to reach its setpoints. 

The Motion commander is really only sending velocity setpoints to the Crazyflie.

The Swarm class handles connection and execution of functions for multiple Crazyflies. It is only a wrapper to remove the need of managing the connections. 

It sounds like a good combination when using the flow deck :-) 

There is no example (I think) with the Swarm class and the MotionCommander, but you can probably look at one of the other swarm examples for some inspiration. For instance, if you look at [swarmSequence.py](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/swarm/swarmSequence.py), you can see that the [connection handling is done with the Swarm class](https://github.com/bitcraze/crazyflie-lib-python/blob/1c07945e7f93587d9e5afd860258aa304b61555d/examples/swarm/swarmSequence.py#L227-L228). To get the MotionCommander into the mix you could update the [run_sequence method](https://github.com/bitcraze/crazyflie-lib-python/blob/1c07945e7f93587d9e5afd860258aa304b61555d/examples/swarm/swarmSequence.py#L205-L220) to something like:

```python 
    def run_sequence(scf, sequence):
        with MotionCommander(scf) as mc:
            # Move one meter forward
            mc.forward(1)
            # Move one meter back
            mc.back(1)
            # The Crazyflie lands when leaving this "with" section
```

**Reply by knmcguire** [reply]
Hi! So you marked Kirstoffers comment as an answer so it is a bit confusing to us if we need to reply to this or not... if it is not resolved yet, could you make a new discussion?

**Reply by Vrinda73** [reply]
Apologies!
